From 0c6f608a682d2ba1e28e7d0f54186e78a44916f4 Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Fri, 5 Aug 2022 23:31:04 -0500
Subject: [PATCH 15/68] sunxi: Globally enable SUPPORT_SPL

This was already supported by every machine type. It is unlikely that
any new SoC support will be added without SPL support.

Signed-off-by: Samuel Holland <samuel@sholland.org>
---
 arch/arm/mach-sunxi/Kconfig | 14 --------------
 board/sunxi/Kconfig         |  2 ++
 2 files changed, 2 insertions(+), 14 deletions(-)

--- a/arch/arm/mach-sunxi/Kconfig
+++ b/arch/arm/mach-sunxi/Kconfig
@@ -184,7 +184,6 @@ config SUN50I_GEN_H6
 	select FIT
 	select SPL_LOAD_FIT
 	select MMC_SUNXI_HAS_NEW_MODE
-	select SUPPORT_SPL
 	---help---
 	Select this for sunxi SoCs which have H6 like peripherals, clocks
 	and memory map.
@@ -225,7 +224,6 @@ config MACH_SUNXI_H3_H5
 	select SUNXI_DRAM_DW
 	select SUNXI_DRAM_DW_32BIT
 	select SUNXI_GEN_SUN6I
-	select SUPPORT_SPL
 
 # TODO: try out A80's 8GiB DRAM space
 config SUNXI_DRAM_MAX_SIZE
@@ -242,7 +240,6 @@ config MACH_SUNIV
 	bool "suniv (Allwinner F1C100s/F1C200s/F1C600/R6)"
 	select CPU_ARM926EJS
 	select SUNXI_GEN_SUN6I
-	select SUPPORT_SPL
 	select SKIP_LOWLEVEL_INIT_ONLY
 	select SPL_SKIP_LOWLEVEL_INIT_ONLY
 
@@ -251,7 +248,6 @@ config MACH_SUN4I
 	select CPU_V7A
 	select DRAM_SUN4I
 	select SUNXI_GEN_SUN4I
-	select SUPPORT_SPL
 	imply SPL_SYS_I2C_LEGACY
 	imply SYS_I2C_LEGACY
 
@@ -260,7 +256,6 @@ config MACH_SUN5I
 	select CPU_V7A
 	select DRAM_SUN4I
 	select SUNXI_GEN_SUN4I
-	select SUPPORT_SPL
 	imply SPL_SYS_I2C_LEGACY
 	imply SYS_I2C_LEGACY
 
@@ -275,7 +270,6 @@ config MACH_SUN6I
 	select SPL_I2C
 	select SUN6I_PRCM
 	select SUNXI_GEN_SUN6I
-	select SUPPORT_SPL
 	select SYS_I2C_SUN6I_P2WI
 	select ARMV7_BOOT_SEC_DEFAULT if OLD_SUNXI_KERNEL_COMPAT
 
@@ -288,7 +282,6 @@ config MACH_SUN7I
 	select SPL_ARMV7_SET_CORTEX_SMPEN
 	select DRAM_SUN4I
 	select SUNXI_GEN_SUN4I
-	select SUPPORT_SPL
 	select ARMV7_BOOT_SEC_DEFAULT if OLD_SUNXI_KERNEL_COMPAT
 	imply SPL_SYS_I2C_LEGACY
 	imply SYS_I2C_LEGACY
@@ -302,7 +295,6 @@ config MACH_SUN8I_A23
 	select DRAM_SUN8I_A23
 	select SPL_I2C
 	select SUNXI_GEN_SUN6I
-	select SUPPORT_SPL
 	select SYS_I2C_SUN8I_RSB
 	select ARMV7_BOOT_SEC_DEFAULT if OLD_SUNXI_KERNEL_COMPAT
 
@@ -315,7 +307,6 @@ config MACH_SUN8I_A33
 	select DRAM_SUN8I_A33
 	select SPL_I2C
 	select SUNXI_GEN_SUN6I
-	select SUPPORT_SPL
 	select SYS_I2C_SUN8I_RSB
 	select ARMV7_BOOT_SEC_DEFAULT if OLD_SUNXI_KERNEL_COMPAT
 
@@ -327,7 +318,6 @@ config MACH_SUN8I_A83T
 	select SUNXI_GEN_SUN6I
 	select MMC_SUNXI_HAS_NEW_MODE
 	select MMC_SUNXI_HAS_MODE_SWITCH
-	select SUPPORT_SPL
 	select SYS_I2C_SUN8I_RSB
 
 config MACH_SUN8I_H3
@@ -347,7 +337,6 @@ config MACH_SUN8I_R40
 	select ARCH_SUPPORT_PSCI
 	select SUNXI_GEN_SUN6I
 	select MMC_SUNXI_HAS_NEW_MODE
-	select SUPPORT_SPL
 	select SUNXI_DRAM_DW
 	select SUNXI_DRAM_DW_32BIT
 	imply SPL_SYS_I2C_LEGACY
@@ -374,7 +363,6 @@ config MACH_SUN8I_V3S
 	select SUNXI_GEN_SUN6I
 	select SUNXI_DRAM_DW
 	select SUNXI_DRAM_DW_16BIT
-	select SUPPORT_SPL
 	select ARMV7_BOOT_SEC_DEFAULT if OLD_SUNXI_KERNEL_COMPAT
 
 config MACH_SUN9I
@@ -385,7 +373,6 @@ config MACH_SUN9I
 	select SPL_I2C
 	select SUN6I_PRCM
 	select SUNXI_GEN_SUN6I
-	select SUPPORT_SPL
 
 config MACH_SUN50I
 	bool "sun50i (Allwinner A64)"
@@ -394,7 +381,6 @@ config MACH_SUN50I
 	select SUNXI_DE2
 	select SUNXI_GEN_SUN6I
 	select MMC_SUNXI_HAS_NEW_MODE
-	select SUPPORT_SPL
 	select SUNXI_DRAM_DW
 	select SUNXI_DRAM_DW_32BIT
 	select FIT
--- a/board/sunxi/Kconfig
+++ b/board/sunxi/Kconfig
@@ -17,6 +17,7 @@ config BOARD_SUNXI
 	select SPL_SEPARATE_BSS if SPL
 	select SUNXI_GPIO if GPIO
 	select SYS_NS16550 if SERIAL
+	select SUPPORT_SPL
 	select SYS_RELOC_GD_ENV_ADDR
 	select USB if DISTRO_DEFAULTS
 	select USB_KEYBOARD if DISTRO_DEFAULTS && USB_HOST
@@ -32,6 +33,7 @@ config BOARD_SUNXI
 	imply FIT
 	imply OF_LIBFDT_OVERLAY
 	imply PRE_CONSOLE_BUFFER
+	imply SPL
 	imply SPL_GPIO
 	imply SPL_LIBCOMMON_SUPPORT
 	imply SPL_LIBGENERIC_SUPPORT
